daemon/graphdriver/zfs/zfs.go

```diff
@@ -271,22 +267,55 @@ func (d *Driver) Create(id string, parent string, mountLabel string, storageOpt
 	}
 
 	// retry
-	return d.create(id, parent)
+	return d.create(id, parent, storageOpt)
 }
 
-func (d *Driver) create(id, parent string) error {
+func (d *Driver) create(id, parent string, storageOpt map[string]string) error {
 	name := d.zfsPath(id)
+	quota, err := parseStorageOpt(storageOpt)
```

Comment:
Perhaps return early on error (also below), because there's now a lot of nested if's (but that's just a suggestion)